src/OgFieldBase.php

```diff
@@ -29,9 +68,21 @@
    * @param String $entity_type
    *   The entity type.
    *
-   * @return OgFieldBase
+   * @return \Drupal\og\OgFieldBase
+   * @throws \Exception
+   *   Throw error if the field storage config definition explicitly defines to
+   *   which entities the field can be attached to.
    */
   public function setEntityType($entity_type) {
+    $field_storage = $this->getFieldStorageConfigBaseDefinition();
+
+    if (!empty($field_storage['entity']) && !in_array($entity_type, $field_storage['entity'])) {
+      // @todo: We need to make sure the field name is being set before the
```

Comment:
I see two options:
1. instead of showing the field name, show the field identifier (i.e. "foo_name" vs "og_group_ref") - it's less descriptive for the poor developer who will see the error, but it will 
2. Try to be nice and show the field name if exists, otherwise the field identifier.

I tend to go with 2.